src/main/java/com/mitchej123/hodgepodge/Compat.java

```diff
@@ -0,0 +1,32 @@
+package com.mitchej123.hodgepodge;
+
+import codechicken.nei.NEIClientConfig;
+import cpw.mods.fml.common.Loader;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+
+/** This class cannot be used earlier than pre-init phase. */
+public class Compat {
+    private static boolean isNeiPresent;
+    private static boolean doesNeiHaveBookmarkAPI;
+
+    static void init() {
+        isNeiPresent = Loader.isModLoaded("NotEnoughItems");
+        try {
+            Method isBookmarkPanelHiddenMethod =
+                    Class.forName("codechicken.nei.NEIClientConfig").getMethod("isBookmarkPanelHidden");
+            if (Modifier.isStatic(isBookmarkPanelHiddenMethod.getModifiers())
+                    && isBookmarkPanelHiddenMethod.getReturnType().equals(boolean.class)) {
+                doesNeiHaveBookmarkAPI = true;
+            }
+        } catch (Exception e) {
+        }
+    }
+
+    public static boolean isNeiLeftPanelVisible() {
+        return isNeiPresent
+                && NEIClientConfig.isEnabled()
```

Comment:
Have you tested without NEI? I'm not sure if Java will throw a ClassNotFoundException here, even if the method call is never reached, simply because this class references `codechicken.nei.NEIClientConfig`.

Reply:
This works just fine. Classes are not loaded until actually being called.

Reply:
Yes. In regular Java code, class references are resolved lazily. (In Mixin code they're resolved ahead of time by the preprocessor though, so the same code *would* cause a crash. This is part of the reason I moved this part of the code to a non-mixin class).